### PR TITLE
Adding pipelines e2e tests from console to console plugin

### PR DESCRIPTION
**Pull request by sanketpathak**
Issue: https://issues.redhat.com/browse/ODC-7592
Story:[ODC-7592](https://issues.redhat.com/browse/ODC-7592)

Description:
Adding e2e test files in the dynamic plugin from the static plugin

Run a test file successfully

Command to execute:

export NO_HEADLESS=true && export CHROME_VERSION=$(/usr/bin/google-chrome-stable --version)
BRIDGE_KUBEADMIN_PASSWORD=YH3jN-PRFT2-Q429c-5KQDr
BRIDGE_BASE_ADDRESS=https://console-openshift-console.apps.dev-svc-4.13-042801.devcluster.openshift.com/
export BRIDGE_KUBEADMIN_PASSWORD
export BRIDGE_BASE_ADDRESS
oc login -u kubeadmin -p $BRIDGE_KUBEADMIN_PASSWORD
oc apply -f ./console/frontend/integration-tests/data/htpasswd-secret.yaml
oc patch oauths cluster --patch "$(cat ./console/frontend/integration-tests/data/patch-htpasswd.yaml)" --type=merge
./test-prow-e2e.sh
Run create-from-add-options.feature
Browser
Chrome 120

### CODE REVIEW

**Review by vikram-raj — COMMENTED** [review]
There is lint error. You can check it by running `yarn lint`

`✖ 85 problems (85 errors, 0 warnings)`

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ integration-tests/screenshots
 yarn-error.log
 .DS_Store
 *.pem
+/integration-tests/cypress-a11y-report.json
```
Comment (ppitonak) [inline .gitignore:10]: leading `/` is unnecessary

File: integration-tests/features/pipelines/create-from-add-options.feature [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+             Then git url "<git_url>" gets Validated
+              And user is able to see info message "<message>" in Pipelines section
+
+        Examples:
+                  | git_url                                | builder_image | message                                                                         |
+                  | https://github.com/sclorg/httpd-ex.git | httpd         | There are no pipeline templates available for Httpd and Deployment combination. |
+                  | https://github.com/sclorg/nginx-ex.git | Nginx         | There are no pipeline templates available for Nginx and Deployment combination. |
+
+
+        @regression @odc-6372
```
Comment (ppitonak) [inline integration-tests/features/pipelines/create-from-add-options.feature:158]:
```
$ yarn gherkin-lint
yarn run v1.22.22
$ ./node_modules/.bin/gherkin-lint -c ./integration-tests/.gherkin-lintrc ./integration-tests/features
/home/ppitonak/workspace/console-plugin/integration-tests/features/pipelines/create-from-add-options.feature
  158    Not allowed tag @odc-6372 on Scenario Outline    allowed-tags
  176    Not allowed tag @odc-6372 on Scenario Outline    allowed-tags
  195    Not allowed tag @odc-7128 on Scenario Outline    allowed-tags
```

Comment (ppitonak) [reply integration-tests/features/pipelines/create-from-add-options.feature:158]: What does `odc-6372` stand for? Jira story?

**Review by ppitonak — COMMENTED** [review]
`yarn install` shows a lot of warnings

```
$ yarn install
yarn install v1.22.22
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning " > @cypress/webpack-preprocessor@5.17.1" has unmet peer dependency "@babel/preset-env@^7.0.0".
warning " > react-i18next@11.18.6" has unmet peer dependency "i18next@>= 19.0.0".
warning " > @openshift/dynamic-plugin-sdk-utils@4.1.0" has unmet peer dependency "@patternfly/react-styles@^5.1.0".
warning " > @openshift/dynamic-plugin-sdk-utils@4.1.0" has unmet peer dependency "react-redux@^7.2.2".
warning " > @openshift/dynamic-plugin-sdk-utils@4.1.0" has incorrect peer dependency "react-router-dom@^6.3.0".
warning " > @openshift/dynamic-plugin-sdk-utils@4.1.0" has unmet peer dependency "redux@^4.1.2".
warning " > @openshift/dynamic-plugin-sdk-utils@4.1.0" has unmet peer dependency "redux-thunk@^2.4.1".
warning " > cache-loader@4.1.0" has incorrect peer dependency "webpack@^4.0.0".
warning " > cypress-axe@0.12.2" has unmet peer dependency "axe-core@^3 || ^4".
warning " > cypress-axe@0.12.2" has incorrect peer dependency "cypress@^3 || ^4 || ^5 || ^6".
warning " > cypress-multi-reporters@1.6.4" has unmet peer dependency "mocha@>=3.1.2".
warning " > formik@2.0.3" has incorrect peer dependency "react@^16.8.0".
warning " > mocha-junit-reporter@2.2.1" has unmet peer dependency "mocha@>=2.2.5".
warning " > mochawesome@7.1.3" has unmet peer dependency "mocha@>=7".
[4/4] Building fresh packages...
Done in 21.22s.
```

File: integration-tests/cypress.config.js [hunk 3]
```diff
@@ -11,7 +17,11 @@ module.exports = defineConfig({
     configFile: 'reporter-config.json',
   },
   fixturesFolder: 'fixtures',
-  defaultCommandTimeout: 5000,
+  chromeWebSecurity: false,
+  env: {
+    TAGS: '@pipelines and (@pre-condition or @smoke or @regression) and not (@manual or @to-do or @un-verified or @broken-test)',
```
Comment (ppitonak) [inline integration-tests/cypress.config.js:22]: I understand why the tag `@pipelines` was needed in the past but do we still need it?

**Comment by ppitonak** [comment]
```
$ yarn test-cypress-headless
yarn run v1.22.22
$ cd integration-tests && node --max-old-space-size=4096 node_modules/.bin/cypress run --env openshift=true --browser ${BRIDGE_E2E_BROWSER_NAME:=chrome} --headless cypress:cli
node:internal/modules/cjs/loader:1146
  throw err;
  ^

Error: Cannot find module '/home/ppitonak/workspace/console-plugin/integration-tests/node_modules/.bin/cypress'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.12.2
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: integration-tests/features/pipelines/create-from-add-options.feature (hunk 2 above)
Comment (sanketpathak) [reply integration-tests/features/pipelines/create-from-add-options.feature:158]: Yes, it's for the respective JIRA story. I can remove these.

File: package.json [hunk 4]
```diff
@@ -47,10 +50,16 @@
     "@types/react-router-dom": "^5.3.2",
     "@typescript-eslint/eslint-plugin": "^5.14.0",
     "@typescript-eslint/parser": "^5.14.0",
+    "babel-loader": "^8.2.1",
+    "babel-plugin-transform-imports": "1.5.1",
     "cache-loader": "^4.1.0",
     "copy-webpack-plugin": "^6.4.1",
     "css-loader": "^6.7.1",
     "cypress": "^11.0.1",
```
Comment (ppitonak) [inline package.json:58]: any reason to use version 11 instead of 13?

Comment (sanketpathak) [reply package.json]: It was already defined earlier, I had this change in my local now updated

**Comment by vikram-raj** [comment]
@sanketpathak Can you resolve the conflicts and update the PR?

**Comment by sanketpathak** [comment]
Closing it in favor of pr https://github.com/openshift-pipelines/console-plugin/pull/133

**Pull request closed without merging**